static/scripts/rendering/plugin-select.ts

```diff
@@ -0,0 +1,93 @@
+import { ManifestCache, ManifestPreDecode, Plugin } from "../../types/plugins";
+import { createElement } from "../../utils/element-helpers";
+import { STRINGS } from "../../utils/strings";
+import { ManifestRenderer } from "../render-manifest";
+import { renderConfigEditor } from "./config-editor";
+import { controlButtons } from "./control-buttons";
+import { closeAllSelect, updateGuiTitle } from "./utils";
+
+export function renderPluginSelector(renderer: ManifestRenderer): void {
+  renderer.currentStep = "pluginSelector";
+  renderer.backButton.style.display = "block";
+  renderer.manifestGuiBody.innerHTML = null;
+  controlButtons({ hide: true });
+
+  const manifestCache = JSON.parse(localStorage.getItem("manifestCache") || "{}") as ManifestCache;
```

Comment:
You have `JSON.parse(localStorage.getItem("manifestCache") || "{}")` more than once, perhaps write a getter function.